### PR TITLE
Added tests to determine when the Servlet filters have modified the response.

### LINKED ISSUE #182: Add tests for Servlet Filter objects.
The following Servlet spec components exist in the project. In anticipation of the porting task #116  add some tests to detect if functionality is the same after the port.

- RouteFilter
- UserTeamsFilter
- ~GrafanaUserFilter~ Grafana disabled in testsuite

### PR DESCRIPTION
**Pull request by whitingjr**
This PR fixes #182 needed for porting to start.

### CODE REVIEW

File: horreum-backend/src/test/java/io/hyperfoil/tools/horreum/server/RouteFilterTest.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+@QuarkusTest
+@QuarkusTestResource(PostgresResource.class)
+@QuarkusTestResource(OidcWiremockTestResource.class)
+@TestProfile(NoGrafanaProfile.class)
+public class RouteFilterTest extends BaseServiceTest {
+
+   @org.junit.jupiter.api.Test
+   public void testRouteRedirectsNotFoundPageToHome() {
+      bareRequest().get("/duff-beer").then()
+         .statusCode(200);
```
Comment (whitingjr) [inline horreum-backend/src/test/java/io/hyperfoil/tools/horreum/server/RouteFilterTest.java:20]: running this test locally the assertion passes.

File: horreum-backend/src/test/java/io/hyperfoil/tools/horreum/server/UserTeamsFilterTest.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+   @org.junit.jupiter.api.Test
+   public void testFilterCreatedLoggedInUser() {
+      String username = "alice";
+      UserInfo userInfo = getUserInfo(username, PESSIMISTIC_WRITE);
+      assertNull(userInfo);
+      bareRequest().cookie("horreum.teams="+TESTER_ROLES[0]).get("/").then()
+         .statusCode(200)
+         .assertThat().cookie("horreum.teams", username + "!" + TESTER_ROLES[0] );
+      UserInfo createdUserInfo = getUserInfo(username, PESSIMISTIC_WRITE);
+      assertNotNull(createdUserInfo);
```
Comment (whitingjr) [inline horreum-backend/src/test/java/io/hyperfoil/tools/horreum/server/UserTeamsFilterTest.java:43]: on a local environment this assertion fails.

**Comment by whitingjr** [comment]
The commit [f0dbed2](https://github.com/Hyperfoil/Horreum/commit/f0dbed29e82e3c961ee3cb540cc257f0c706eded) is missing tests that check GrafanaFilter.

File: horreum-backend/src/test/java/io/hyperfoil/tools/horreum/server/RouteFilterTest.java (hunk 1 above)
Comment (johnaohara) [reply horreum-backend/src/test/java/io/hyperfoil/tools/horreum/server/RouteFilterTest.java:24]: DYK this test is failing in CI env? I would be cautious about disabling tests in CI as we are then relying on contributors running the testsuite locally and we don't have full coverage in CI.

Comment (whitingjr) [reply horreum-backend/src/test/java/io/hyperfoil/tools/horreum/server/RouteFilterTest.java:24]: Yes it was failing. Along with tests in UserTeamsFilterTest.
 Checking the logs produced by CI today shows test case is now skipped.

Comment (whitingjr) [reply horreum-backend/src/test/java/io/hyperfoil/tools/horreum/server/RouteFilterTest.java:24]: having configured the connection information these are now working.

**Comment by johnaohara** [comment]
> @johnaohara I am seeing 401 rc for requests with an oauth token. I was expecting 200. Is that wrong ?

I don't how that can work, as the client is sending a client generated JWT token, and the service will be expecting a keycloak generated oauth2 access token. I would have thought you would have had to generate an access token from keyloak to pass into the Reastassured call

**Comment by johnaohara** [comment]
@whitingjr with this change: https://github.com/johnaohara/Horreum/commit/af6604634905d04f202a02d1a3642ee9baa962a2
The test now authenticates, but is returning a http 404 error: https://github.com/johnaohara/Horreum/runs/6511551184?check_suite_focus=true

**Comment by whitingjr** [comment]
that's odd. as now I am seeing the tests pass on my local env.
```
$ mvn test  -Dnpm.skip  -DfailIfNoTests=false  -DCOMPOSE_OVERRIDE=podman-compose
...
[INFO] Tests run: 2, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 104.298 s - in io.hyperfoil.tools.horreum.server.GrafanaUserFilterTest
...
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------

```

**Pull request closed without merging**